### PR TITLE
Attempt to fix method calling in animation tree

### PR DESCRIPTION
**Pull request by IllusiveS**
This is my first pull request for open source so be gentle.
Fix for #25950
As i understand the system, caches applying to the same path are overriding each other since they are being held in a hash map, so duplicates are not working. 
My solution is to make cache a hashmap of NodePath and set of TrackCache.

### CODE REVIEW

**Comment by KoBeWi** [comment]
You linked wrong issue.

**Comment by IllusiveS** [comment]
> You linked wrong issue.

Fixed

**Comment by IllusiveS** [comment]
So do think that issue #25950 is fixed
Amusingly i still can't call a method from animation in my project so i seem to have a different problem

**Comment by akien-mga** [comment]
Could you squash commits together, and edit the resulting commit log to be more descriptive?
See http://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html and https://github.com/godotengine/godot/blob/master/CONTRIBUTING.md#format-your-commit-logs-with-readability-in-mind

**Comment by akien-mga** [comment]
I don't think turning this cache into a set is the correct solution. The actual cause for the bug seems to be the one described at https://github.com/godotengine/godot/issues/25950#issuecomment-473758324

Since different function tracks share the same (empty) name, they override each other. So I think the proper fix would be to make sure that each track has a unique name.

File: scene/animation/animation_tree.cpp [hunk 1]
```diff
@@ -860,243 +891,206 @@ void AnimationTree::_process_graph(float p_delta) {
 
 				ERR_CONTINUE(!track_cache.has(path));
```
Comment (aclave1) [inline scene/animation/animation_tree.cpp]: I'm seeing this error firing frequently

**Comment by aclave1** [comment]
Not sure why, but when patched on top of f4e370189 this PR breaks my animation tree. All characters are TPosed. Animation player on importing a GLB works fine but when I add an animation tree, things start breaking.

**Comment by aaronfranke** [comment]
@IllusiveS Is this still desired? If so, it needs to be rebased on the latest master branch, and the feedback should be addressed.

If not, abandoned pull requests will be closed in the future as announced [here](https://godotengine.org/article/headsup-vulkan-merged-master-branch-unstable).

**Comment by aclave1** [comment]
@aaronfranke #25950 was closed, does this PR address a separate issue?

**Comment by aaronfranke** [comment]
Oh, I see. Closing as fixed by a mysterious unknown commit :eyes:

**Pull request closed without merging**